### ISSUE TITLE
Memory issue crnn

### ISSUE THREAD

**Issue by NikitaPomies** [mindee/doctr#1356]
### Bug description

I am running this function :
![Capture d’écran du 2023-10-18 22-58-36](https://github.com/mindee/doctr/assets/83075701/6b706aa3-7144-4568-95d4-8a67f068034d)
with the model being  load with the following code :

      `        

        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
        # self.device = torch.device("cpu")

        det_model = db_resnet50(pretrained=False, pretrained_backbone=False)
        det_params = torch.load(path_model_detection, map_location=self.device)
        det_model.load_state_dict(det_params)
        reco_model = crnn_vgg16_bn(pretrained=False, pretrained_backbone=False)
        reco_params = torch.load(path_model_recognition, map_location=self.device)
        reco_model.load_state_dict(reco_params)

        self.model = ocr_predictor(
            det_arch=det_model,
            reco_arch=reco_model,
            pretrained=False,
            assume_straight_pages=True,
        ).to(self.device)
        
 After investigating, the leak seems to come from the crnn class, as if the feat_extractor was still computing the gradients  :
        
![Capture d’écran du 2023-10-18 22-59-26](https://github.com/mindee/doctr/assets/83075701/b8cc9d95-3ecf-41e5-888a-6b946cde2367)

The memory is never freed and running ocr_image multiple times results in a huge increase of ram usage.

Do someone have any insights on that ? 
I am using last doctr main version and torch==1.12.0+cpu on Linux-5.15.0-58-generic-x86_64-with-glibc2.35

### Code snippet to reproduce the bug

"""

### Error traceback

"""

### Environment

"""

### Deep Learning backend

is_tf_available: False
is_torch_available: True

**Comment by felixdittrich92** [comment]
Hi @NikitaPomies :wave:,

Thanks for the report.
I was able to reproduce the behaviour and will open a PR to fix this soon. :+1:

**Comment by NikitaPomies** [comment]
> Hi @NikitaPomies :wave:,
> 
> 
> 
> Thanks for the report.
> 
> I was able to reproduce the behaviour and will open a PR to fix this soon. :+1: 

Thanks ! 
Just for the record,
I have been able to avoid the high memory usage by putting 

ONEDNN_PRIMITIVE_CACHE_CAPACITY = 1 

as suggested on https://github.com/pytorch/pytorch/issues/29893

**Comment by felixdittrich92** [comment]
Thanks for the response :+1:
